SpatialGDK/Source/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -51,9 +51,36 @@ bool USpatialGameInstance::HasSpatialNetDriver() const
 	return bHasSpatialNetDriver;
 }
 
+// Delegate used in Engine.h/UnrealEngine.cpp to allow client travel in a Spatial context.
+// Will initialize the PendingNetGame and the NetDriver to Spatial variants.
+void SpatialBrowse(FWorldContext& WorldContext, FURL URL)
+{
+	bool bIsSpatial = false;
+	UNetDriver* TempNetDriver = nullptr;
+	if (GEngine->CreateNamedNetDriver(WorldContext.PendingNetGame, NAME_PendingNetDriver, NAME_GameNetDriver))
+	{
+		TempNetDriver = GEngine->FindNamedNetDriver(WorldContext.PendingNetGame, NAME_PendingNetDriver);
+		// Setting the PendingNetGame's NetDriver is necessary here because the call to CreateNamedNetDriver above will interfere
+		// with the internals of InitNetDriver and cause the NetDriver not to be initialized, and fail a check().
+		WorldContext.PendingNetGame->NetDriver = TempNetDriver;
+		bIsSpatial = GetDefault<UGeneralProjectSettings>()->bSpatialNetworking;
```

Comment:
This line removes the necessity for this entire if block. The only reason I instantiated the net driver here was to check if it was a `USpatialNetDriver` and set the value of `bIsSpatial`. I think you can now refactor this function into just a few lines:

```
if (GetDefault<UGeneralProjectSettings>()->bSpatialNetworking)
{
  WorldContext.PendingNetGame = NewObject<USpatialPendingNetGame>();
  WorldContext.PendingNetGame->Initialize(URL);
}
```